src/pages/PlanPage/subRoutes/CurrentOrgPlan/BillingDetails/EmailAddress/EmailAddress.spec.jsx

```diff
@@ -0,0 +1,198 @@
+  defaultOptions: { queries: { suspense: true } },
+})
+const server = setupServer()
+
+const wrapper =
+  (initialEntries = '/plan/gh/codecov') =>
+  ({ children }) =>
+    (
+      <QueryClientProvider client={queryClient}>
+        <MemoryRouter initialEntries={[initialEntries]}>
+          <Route path="/plan/:provider/:owner">
+            <Suspense fallback={null}>{children}</Suspense>
+          </Route>
+        </MemoryRouter>
+      </QueryClientProvider>
+    )
+
+beforeAll(() => {
+  server.listen()
+})
+afterEach(() => {
+  queryClient.clear()
+  server.resetHandlers()
+})
+afterAll(() => {
+  server.close()
+})
+
+const testEmail = 'testemail@gmail.com'
+
+const mockAccountDetails = {
+  subscriptionDetail: {
+    customer: {
+      email: testEmail,
+    },
+  },
+}
+
+describe('EmailAddress', () => {
+  function setup() {
+    const user = userEvent.setup()
+    const mutate = jest.fn()
+
+    server.use(
+      rest.get('/internal/gh/codecov/account-details', (req, res, ctx) =>
+        res(ctx.status(200), ctx.json(mockAccountDetails))
+      ),
+      rest.patch(
+        '/internal/gh/codecov/account-details/update_email',
+        (req, res, ctx) => {
+          mutate()
+          return res(ctx.status(200), ctx.json({}))
+        }
+      )
+    )
+
+    return { user, mutate }
+  }
+
+  describe(`when the form is closed`, () => {
+    beforeEach(() => setup())
+
+    it('shows the email field', async () => {
+      render(<EmailAddress />, { wrapper: wrapper() })
+
+      const currentEmail = await screen.findByText('testemail@gmail.com')
+      expect(currentEmail).toBeInTheDocument()
+    })
+
+    it('shows the email title field', async () => {
+      render(<EmailAddress />, { wrapper: wrapper() })
+
+      const emailTitle = await screen.findByText('Email address')
+      expect(emailTitle).toBeInTheDocument()
+    })
+
+    it('shows the edit email button', async () => {
+      render(<EmailAddress />, { wrapper: wrapper() })
+
+      const editButton = await screen.findByTestId('edit-email')
+      expect(editButton).toBeInTheDocument()
+    })
+  })
+
+  describe('when the user clicks on Edit card', () => {
+    it('shows the placeholder email', async () => {
+      const { user } = setup()
+      render(<EmailAddress />, { wrapper: wrapper() })
+
+      const editButton = await screen.findByTestId('edit-email')
+      await user.click(editButton)
+
+      const placeholderEmail = await screen.findByText('Your email')
+      expect(placeholderEmail).toBeInTheDocument()
+    })
+
+    it('shows the update button', async () => {
+      const { user } = setup()
+      render(<EmailAddress />, { wrapper: wrapper() })
+
+      const editButton = await screen.findByTestId('edit-email')
+      await user.click(editButton)
+
+      const updateButton = screen.getByRole('button', { name: /Update/i })
+      expect(updateButton).toBeInTheDocument()
+    })
+
+    it('shows the cancel button', async () => {
```

Comment:
Can we add a test for clicking the button + closing the form please?

Reply:
Yeap, just noticed I missed that one 😬